jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -971,8 +971,6 @@ protected void onCompleteSuccess()
         public void onCompleteFailure(final Throwable x)
         {
             failedCallback(release(), x);
-            if (_shutdownOut)
-                getEndPoint().shutdownOutput();
```

Comment:
Why was this removed? Is it done elsewhere?
We want to gently close in case of TLS.

Reply:
@sbordet  This was done by @joakime before I "hijacked" the branch.  I think it was done during a collective debugging sessions, resolving the initial race condition, where one thread was shutting does the endpoint whilst another was trying to write out the error page.

Can you more precisely describe the use-case you are concerned about and I'll double check that we test for it and see who is doing the shutdown.

Reply:
The `SHUTDOWN` result is only returned after successful writes (and therefore another iteration on the generator), so I guess it's only a concern on the succeeded path.

I think it's done in 2 places now, `SendCallback.onCompleteSuccess()` and then again in `HTTPStreamOverHTTP1.succeeded()`, so I think the former can be removed.

For the failure case we should always end up in `HTTPStreamOverHTTP1.failed()` so we `abort()` and that should be fine.

Reply:
done.